pkg/controller/controller.go

```diff
@@ -631,7 +617,7 @@ func NewClientConfigurationForBroker(broker *v1beta1.ClusterServiceBroker, authC
 // reconciliationRetryDurationExceeded returns whether the given operation
 // start time has exceeded the controller's set reconciliation retry duration.
 func (c *controller) reconciliationRetryDurationExceeded(operationStartTime *metav1.Time) bool {
-	if time.Now().Before(operationStartTime.Time.Add(c.reconciliationRetryDuration)) {
+	if operationStartTime == nil || time.Now().Before(operationStartTime.Time.Add(c.reconciliationRetryDuration)) {
```

Comment:
I was interpreting it in the sense that: if its operation start time is `nil`, then no operation has started, so its retry duration couldn't have been exceeded, and this would return `false`. I think it's a little hacky to say "you don't have an operation start time, but I know you'd only call this function if you thought you did, so I'm going to say you expired so you abort anyway, even if the reason why isn't quite accurate."

I think the real question is then whether it's okay to remove the guard on `OperationStartTime != nil` from the polling function. I admit I did it because it was additional logic and a lot of LoC for little benefit in my mind. That would imply bugs in both the API server and the controller, and absolute worst-case the resource would just poll forever with a clearly invalid `Status`. I think, short of situations where the controller would panic, it's alright for the controller to assume the integrity of API server resources.

If you feel strongly about it, though, I'm okay with adding the guard back in.

Reply:
I sounds hacking when you phrase it that way. If instead you phrase yours as "you don't have an operation start time, I don't know when you started your current operation, but i'm just going to assume that you haven't expired even though I don't really have any basis for that", then yours sounds hacky to me."

As for the guard in the polling function, it was only there because there was complaints about sprinkling the `OperationStartTime != nil` check into all of the places in the polling function where the `OperationStartTime` was being checked. With the changes made to move the logic for checking and processing reconciliation retry timeout, there is no reason to have the guard on the polling function any more.

I do agree that the impact of *not* stopping reconciliation after the retry duration should have expired is not catastrophic. I don't feel strongly enough about the behavior of `reconciliationRetryDurationExceeded` to require a certain behavior if you feel strongly the other way.

Reply:
Ah, okay, didn't realize that was the motivation for the guard! I think it's personally just a little cleaner the current way, since when an operation is not going on, an instance's `OperationStartTime` will in fact be `nil`, so in case we wanted to use the function in a situation where an operation hasn't necessarily started, it'd still function as expected.